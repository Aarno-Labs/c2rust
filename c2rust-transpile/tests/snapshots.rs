use std::fs;

#[test]
fn transpile() {
    let config = || c2rust_transpile::TranspilerConfig {
        dump_untyped_context: false,
        dump_typed_context: false,
        pretty_typed_context: false,
        dump_function_cfgs: false,
        json_function_cfgs: false,
        dump_cfg_liveness: false,
        dump_structures: false,
        verbose: false,
        debug_ast_exporter: false,
        incremental_relooper: true,
        fail_on_multiple: false,
        filter: None,
        debug_relooper_labels: false,
        prefix_function_names: None,
        translate_asm: true,
        use_c_loop_info: true,
        use_c_multiple_info: true,
        simplify_structures: true,
        panic_on_translator_failure: false,
        emit_modules: false,
        fail_on_error: false,
        replace_unsupported_decls: c2rust_transpile::ReplaceMode::Extern,
        translate_valist: true,
        overwrite_existing: true,
        reduce_type_annotations: false,
        reorganize_definitions: false,
        enabled_warnings: Default::default(),
        emit_no_std: false,
        output_dir: None,
        translate_const_macros: false,
        translate_fn_macros: false,
        disable_refactoring: false,
        preserve_unused_functions: false,
        log_level: log::LevelFilter::Warn,
        emit_build_files: false,
        binaries: Vec::new(),
    };
    insta::glob!("snapshots/*.c", |f: &Path| {
        let (_temp_dir, temp_path) =
            c2rust_transpile::create_temp_compile_commands(&[f.to_owned()]);
        c2rust_transpile::transpile(config(), &temp_path, &[]);
        let output_path = f.with_extension("rs");
        let output = fs::read_to_string(&output_path).unwrap();
        fs::remove_file(&output_path).unwrap();
        insta::assert_snapshot!(&output);
    });
}
