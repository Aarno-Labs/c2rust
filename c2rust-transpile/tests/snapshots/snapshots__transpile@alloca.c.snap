---
source: c2rust-transpile/tests/snapshots.rs
expression: cat tests/snapshots/alloca.rs
input_file: c2rust-transpile/tests/snapshots/alloca.c
---
#![allow(
    dead_code,
    mutable_transmutes,
    non_camel_case_types,
    non_snake_case,
    non_upper_case_globals,
    unused_assignments,
    unused_mut
)]
#[no_mangle]
pub static mut TRUE: std::ffi::c_int = 1 as std::ffi::c_int;
#[no_mangle]
pub unsafe extern "C" fn alloca_sum(
    mut val1: std::ffi::c_int,
    mut val2: std::ffi::c_int,
) -> std::ffi::c_int {
    let mut alloca_allocations: Vec<Vec<u8>> = Vec::new();
    let mut alloca1: *mut std::ffi::c_int = 0 as *mut std::ffi::c_int;
    let mut alloca2: *mut std::ffi::c_int = 0 as *mut std::ffi::c_int;
    if TRUE != 0 {
        alloca_allocations.push(::std::vec::from_elem(
            0,
            ::core::mem::size_of::<std::ffi::c_int>() as usize,
        ));
        alloca1 = alloca_allocations.last_mut().unwrap().as_mut_ptr() as *mut std::ffi::c_int;
        *alloca1 = val1;
    }
    if TRUE != 0 {
        alloca_allocations.push(::std::vec::from_elem(
            0,
            ::core::mem::size_of::<std::ffi::c_int>() as usize,
        ));
        alloca2 = alloca_allocations.last_mut().unwrap().as_mut_ptr() as *mut std::ffi::c_int;
        *alloca2 = val2;
    }
    return *alloca1 + *alloca2;
}